crates/worker/src/partition_processor_manager.rs

```diff
@@ -70,14 +74,17 @@ use crate::PartitionProcessorBuilder;
 pub struct PartitionProcessorManager {
     task_center: TaskCenter,
     updateable_config: Live<Configuration>,
-    running_partition_processors: BTreeMap<PartitionId, State>,
+    running_partition_processors: BTreeMap<PartitionId, ProcessorState>,
     name_cache: BTreeMap<PartitionId, &'static str>,
 
     metadata: Metadata,
     metadata_store_client: MetadataStoreClient,
     partition_store_manager: PartitionStoreManager,
     attach_router: RpcRouter<AttachRequest, Networking>,
-    incoming_get_state: BoxStream<'static, MessageEnvelope<GetProcessorsState>>,
+    incoming_get_state:
+        Pin<Box<dyn Stream<Item = MessageEnvelope<GetProcessorsState>> + Send + Sync + 'static>>,
```

Comment:
Curious, why did we need to remove BoxStream stream?

Reply:
I needed to add the `Sync` trait to the `Stream` trait object. That's why I had to spell things out more verbosely :-(